Library/RSBot.Core/Objects/Player.cs

```diff
@@ -609,60 +621,57 @@ public bool MoveTo(Position destination, bool sleep = true)
             return false;
         }
 
-        /// <summary>
-        /// Uses the health potion.
-        /// </summary>
-        public bool UseHealthPotion()
+        private bool UsePotion(TypeIdFilter filter, ref int tick, ref int duration)
         {
             if (State.LifeState == LifeState.Dead)
                 return false;
 
-            var elapsed = Environment.TickCount - _lastHpPotionTick;
-            if (Race == ObjectCountry.Chinese && elapsed < 1000)
-                return false;
+            InventoryItem potionItem = null;
+            if (_lastHPDuration == 0)
+            {
+                potionItem = Inventory.GetItem(filter);
+                if (potionItem == null)
+                    return false;
 
-            if (Race == ObjectCountry.Europe && elapsed < 15000)
-                return false;
+                // potion
+                if (potionItem.Record.Param1 > 0)
+                {
+                    if (Race == ObjectCountry.Chinese)
+                        duration = 1000;
+                    else
+                        duration = 15000;
+                }
+                // grain
+                else if (potionItem.Record.Param2 > 0)
+                {
+                    duration = 4000;
+                }
+            }
 
-            var typeIdFilter = new TypeIdFilter(3, 3, 1, 1);
-            var potionItem = Inventory.GetItem(typeIdFilter);
-            if (potionItem == null)
+            var elapsed = Environment.TickCount - tick;
+            if (elapsed < duration)
                 return false;
 
             var result = potionItem.Use();
```

Comment:
potionItem is null in my server because. Maybe add null check

Reply:
There have already null check. I tested and it was working. Whatever i will try again.